### DISCUSSION TITLE
Using Angstrom ?

### DISCUSSION THREAD

**Discussion by mefyl** [mirage/ocaml-cohttp#1018] [Ideas]
The parser defined in `http` reimplements a minimal version of Angstrom. Was not using Angstrom a conscious choice? It would have several advantages, namely using more battle-tested code, less maintenance work and more interoperability - someone implementing websockets for instance could easily pickup the request headers parser from `http` and combine upon it.

**Comment by rgrinberg** [comment]
It was written by @anuragsoni. From the benchmarks, it had better performance and didn't consume an additional dependency. I'll let him expand if there were any other advantages.

**Reply by anuragsoni** [reply]
@rgrinberg I think you covered it all. The primary motivation behind this was the perf improvements over angstrom, and an easier integration into the existing cohttp library without requiring a larger scale re-write to drive angstrom.
